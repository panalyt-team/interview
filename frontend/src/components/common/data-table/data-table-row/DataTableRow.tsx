import { makeStyles } from "@material-ui/styles";
import React from "react";

export interface DataTableRowProps {
    rowValues: string[];
}

const useStyles = makeStyles({
    root: {
        display: "grid",
        gridTemplateColumns: "repeat(auto-fit, minmax(100px, 1fr))",
        textAlign: "center",
        width: "100%",
        height: "100px",
        justifyContent: "space-around",
        alignItems: "center",
        backgroundColor: "#fafafa",
        color: "#2d333b",
        border: "1px solid #e1e4e8"
    },
    label: { 
        fontSize: "1.2em",
        letterSpacing: "0.03em"
    }
}, { classNamePrefix: "data-table-row"});

export const DataTableRow = ({ rowValues }: DataTableRowProps): JSX.Element => {
    const classes = useStyles();
    return (
        <div className={classes.root}>
            {
                rowValues.map((column: string, index: number) => 
                    <p key={index} className={classes.label}>{column}</p>
                )
            }
        </div>
    )
};

export default DataTableRow;