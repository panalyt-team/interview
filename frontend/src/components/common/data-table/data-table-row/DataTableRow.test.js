import { shallow } from 'enzyme';
import DataTableRow from './DataTableRow';
import { render } from '@testing-library/react';

describe("DataTableRow", () => {

    it("renders correctly", () => {
        const wrapper = shallow(<DataTableRow rowValues={["China", "8 516,87 $", { "label": "+9.12%", "color": "#adff2e" }]} />);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders a chip when there is a ChipProperties prop in rowValues", () => {
        const { container } = render(<DataTableRow rowValues={["China", "8 516,87 $", { "label": "+9.12%", "color": "#adff2e" }]} />);

        const chip = container.querySelector('[data-test-id="chip"]');
        
        expect(chip).toBeTruthy();
    });

    it("does not render a chip when there is no ChipProperties prop in rowValues", () => {
        const { container } = render(<DataTableRow rowValues={["China", "8 516,87 $", "test value"]} />);

        const chip = container.querySelector('[data-test-id="chip"]');
        
        expect(chip).toBeFalsy();
    });

});