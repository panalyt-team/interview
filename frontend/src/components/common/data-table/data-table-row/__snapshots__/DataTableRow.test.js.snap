// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableRow renders correctly 1`] = `ShallowWrapper {}`;
